ngramoffset.go

```diff
@@ -18,6 +18,17 @@ import (
 	"sort"
 )
 
+// shrinkUint32Slice copies slices with excess capacity to precisely seized ones
+// to avoid wasting memory. It should be used on slices with long static durations.
+func shrinkUint32Slice(a []uint32) []uint32 {
```

Comment:
Could we avoid the copy by using https://tip.golang.org/doc/go1.2#three_index? i.e `shrinkUint32Slice(foo) == foo[:len(foo):len(foo)]`

Reply:
> but its pointer will reference and retain the original (too-large) memory allocation.

Can you point me to some docs that say the garbage collector doesn't reclaim this memory?

Reply:
https://blog.golang.org/slices-intro#TOC_6.

Reply:
Thanks!